### PR TITLE
[HMA-2463] - 2020/2021 Update

### PR DESCRIPTION
**Pull request by georgeherby**
* This has the new National Insurance bands
* Data-Driven Tests for testing codes for 2020 going forward
* Upgrade to Kotlin 1.3.70

### CODE REVIEW

File: src/jvmTest/kotlin/Tests2020.kt [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+//    fun `2020 Test Scenarios`() {
+//
+//        listOfTest.forEach { testData: TestData ->
+//            @Test
+//            fun `Title: ${testData.title} TaxCode: ${testData.taxCode}`(string: String) {
+//
+//            }
+//        }
+//    }
+//}
```
Comment (jvanderwee) [inline src/jvmTest/kotlin/Tests2020.kt]:
```suggestion
```

File: src/jvmTest/kotlin/Tests2020.kt [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        TestData(Country.SCOTLAND, "D2 code", 12000.0, "SD2", 5520.0, 300.0, 5820.0, 6180.0, 443.26)
+        ).map {
+
+        dynamicTest("${it.country} - ${it.scenario} - ${it.taxCode} - ${it.salary}") {
+
+            val response: CalculatorResponse =
+                Calculator(it.taxCode, it.salary, payPeriod = PayPeriod.YEARLY, taxYear = 2020).run()
+
+
+//            print(response.yearly.taxBreakdown)
```
Comment (jvanderwee) [inline src/jvmTest/kotlin/Tests2020.kt]:
```suggestion
```

File: src/commonTest/kotlin/uk/gov/hmrc/calculator/model/bands/EmployeeNIBandsTests.kt [hunk 3]
```diff
@@ -63,4 +63,36 @@ class EmployeeNIBandsTests {
         assertEquals(false, band.inBand(1000.0))
         assertEquals(true, band.inBand(100000.0))
     }
+
+    @Test
+    fun `Employee NI 2020 England`() {
```
Comment (jvanderwee) [inline src/commonTest/kotlin/uk/gov/hmrc/calculator/model/bands/EmployeeNIBandsTests.kt]: do we need separate tests for each country if they are using the same bands?

Comment (georgeherby) [reply src/commonTest/kotlin/uk/gov/hmrc/calculator/model/bands/EmployeeNIBandsTests.kt]: Resolved

**Pull request merged**